### ISSUE TITLE
Benchmarks

### ISSUE THREAD

**Issue by whatisinternet** [whatisinternet/inflector#25]
This is to track benchmark results for case conversions performance issues seen in #18.

Obviously there is a huge issue with class case. That said several other areas are quite slow too.

Tests were run on: `mid-2014 MBP i5 2.6`

## Current state of [master](https://github.com/whatisinternet/inflector/tree/14290b8a7f104e19dda1e2e757c13a6f2b34d844)
```shell
test cases::camelcase::tests::bench_camel                                         ... bench:       1,825 ns/iter (+/- 346)
test cases::camelcase::tests::bench_camel_from_sname                              ... bench:       1,223 ns/iter (+/- 289)
test cases::camelcase::tests::bench_is_camel                                      ... bench:      49,416 ns/iter (+/- 593)
test cases::classcase::tests::bench_class_case                                    ... bench: 160,985,376 ns/iter (+/- 5,173,751)
test cases::classcase::tests::bench_class_from_snake                              ... bench: 161,533,425 ns/iter (+/- 4,167,305)
test cases::classcase::tests::bench_is_class                                      ... bench: 161,352,118 ns/iter (+/- 3,793,478)
test cases::kebabcase::tests::bench_is_kebab                                      ... bench:         793 ns/iter (+/- 400)
test cases::kebabcase::tests::bench_kebab                                         ... bench:         752 ns/iter (+/- 310)
test cases::kebabcase::tests::bench_kebab_from_snake                              ... bench:         210 ns/iter (+/- 32)
test cases::lowercase::tests::bench_is_lower                                      ... bench:         340 ns/iter (+/- 86)
test cases::lowercase::tests::bench_lower                                         ... bench:         306 ns/iter (+/- 173)
test cases::screamingsnakecase::tests::bench_is_screaming_snake                   ... bench:         635 ns/iter (+/- 210)
test cases::screamingsnakecase::tests::bench_screaming_snake                      ... bench:         610 ns/iter (+/- 87)
test cases::screamingsnakecase::tests::bench_screaming_snake_from_camel           ... bench:         961 ns/iter (+/- 579)
test cases::screamingsnakecase::tests::bench_screaming_snake_from_class           ... bench:         894 ns/iter (+/- 352)
test cases::screamingsnakecase::tests::bench_screaming_snake_from_kebab           ... bench:         877 ns/iter (+/- 571)
test cases::screamingsnakecase::tests::bench_screaming_snake_from_screaming_snake ... bench:         584 ns/iter (+/- 304)
test cases::screamingsnakecase::tests::bench_screaming_snake_from_sentence        ... bench:       1,123 ns/iter (+/- 630)
test cases::screamingsnakecase::tests::bench_screaming_snake_from_upper_kebab     ... bench:         914 ns/iter (+/- 435)
test cases::sentencecase::tests::bench_is_sentence                                ... bench:       2,714 ns/iter (+/- 796)
test cases::sentencecase::tests::bench_sentence                                   ... bench:       2,678 ns/iter (+/- 1,357)
test cases::sentencecase::tests::bench_sentence_from_snake                        ... bench:       2,100 ns/iter (+/- 1,046)
test cases::snakecase::tests::bench_is_snake                                      ... bench:         626 ns/iter (+/- 191)
test cases::snakecase::tests::bench_snake                                         ... bench:         581 ns/iter (+/- 298)
test cases::snakecase::tests::bench_snake_from_camel                              ... bench:         882 ns/iter (+/- 328)
test cases::snakecase::tests::bench_snake_from_class                              ... bench:         883 ns/iter (+/- 193)
test cases::snakecase::tests::bench_snake_from_kebab                              ... bench:         922 ns/iter (+/- 360)
test cases::snakecase::tests::bench_snake_from_sentence                           ... bench:       1,209 ns/iter (+/- 539)
test cases::snakecase::tests::bench_snake_from_snake                              ... bench:         637 ns/iter (+/- 386)
test cases::snakecase::tests::bench_snake_from_upper_kebab                        ... bench:         876 ns/iter (+/- 488)
test cases::tablecase::tests::bench_is_table_case                                 ... bench:       5,784 ns/iter (+/- 1,129)
test cases::tablecase::tests::bench_table_case                                    ... bench:       5,754 ns/iter (+/- 1,460)
test cases::titlecase::tests::bench_is_title                                      ... bench:       2,847 ns/iter (+/- 1,553)
test cases::titlecase::tests::bench_title                                         ... bench:       2,799 ns/iter (+/- 1,309)
test cases::titlecase::tests::bench_title_from_snake                              ... bench:       2,202 ns/iter (+/- 697)
test cases::uppercase::tests::bench_is_upper                                      ... bench:         357 ns/iter (+/- 55)
test cases::uppercase::tests::bench_upper                                         ... bench:         311 ns/iter (+/- 135)
```

**Comment by whatisinternet** [comment]
[Tracking branch](https://github.com/whatisinternet/inflector/compare/enhancement/performance)

**Comment by whatisinternet** [comment]
[Related card](https://github.com/whatisinternet/inflector/projects/1#card-205372)

**Comment by whatisinternet** [comment]
Current state of [Tracking branch](https://github.com/whatisinternet/inflector/compare/enhancement/performance)

```shell
test cases::camelcase::tests::bench_camel0                      ... bench:         614 ns/iter (+/- 231)
test cases::camelcase::tests::bench_camel1                      ... bench:         628 ns/iter (+/- 301)
test cases::camelcase::tests::bench_camel2                      ... bench:         616 ns/iter (+/- 162)
test cases::camelcase::tests::bench_is_camel                    ... bench:         653 ns/iter (+/- 293)
test cases::classcase::tests::bench_class_case                  ... bench:       3,051 ns/iter (+/- 976)
test cases::classcase::tests::bench_class_from_snake            ... bench:       3,071 ns/iter (+/- 1,125)
test cases::classcase::tests::bench_is_class                    ... bench:       3,195 ns/iter (+/- 2,123)
test cases::kebabcase::tests::bench_is_kebab                    ... bench:         531 ns/iter (+/- 93)
test cases::kebabcase::tests::bench_kebab                       ... bench:         461 ns/iter (+/- 150)
test cases::kebabcase::tests::bench_kebab_from_snake            ... bench:         799 ns/iter (+/- 440)
test cases::lowercase::tests::bench_is_lower                    ... bench:         345 ns/iter (+/- 87)
test cases::lowercase::tests::bench_lower                       ... bench:         302 ns/iter (+/- 164)
test cases::screamingsnakecase::tests::bench_is_screaming_snake ... bench:         494 ns/iter (+/- 95)
test cases::screamingsnakecase::tests::bench_screaming_snake    ... bench:         480 ns/iter (+/- 232)
test cases::sentencecase::tests::bench_is_sentence              ... bench:         687 ns/iter (+/- 300)
test cases::sentencecase::tests::bench_sentence                 ... bench:         632 ns/iter (+/- 225)
test cases::sentencecase::tests::bench_sentence_from_snake      ... bench:         715 ns/iter (+/- 217)
test cases::snakecase::tests::bench_is_snake                    ... bench:         526 ns/iter (+/- 264)
test cases::snakecase::tests::bench_snake_from_camel            ... bench:         693 ns/iter (+/- 187)
test cases::snakecase::tests::bench_snake_from_snake            ... bench:         843 ns/iter (+/- 461)
test cases::snakecase::tests::bench_snake_from_title            ... bench:         466 ns/iter (+/- 136)
test cases::tablecase::tests::bench_is_table_case               ... bench:       2,817 ns/iter (+/- 989)
test cases::tablecase::tests::bench_table_case                  ... bench:       2,776 ns/iter (+/- 1,040)
test cases::titlecase::tests::bench_is_title                    ... bench:         771 ns/iter (+/- 360)
test cases::titlecase::tests::bench_title                       ... bench:         829 ns/iter (+/- 388)
test cases::titlecase::tests::bench_title_from_snake            ... bench:         736 ns/iter (+/- 315)
test cases::uppercase::tests::bench_is_upper                    ... bench:         400 ns/iter (+/- 57)
test cases::uppercase::tests::bench_upper                       ... bench:         393 ns/iter (+/- 249)
```

**Comment by whatisinternet** [comment]
Closed by #26
